File: Editorial/55_Jump_Game/C/1_Backtracking.c
bool canJumpFromPosition(int position, int* nums, int numsSize) {
    if (position == numsSize - 1) {
        return true;
    }
    int furthestJump = position + nums[position] < numsSize - 1
                           ? position + nums[position]
                           : numsSize - 1;
    for (int nextPosition = position + 1; nextPosition <= furthestJump;
         ++nextPosition) {
        if (canJumpFromPosition(nextPosition, nums, numsSize)) {
            return true;
        }
    }
    return false;
}
bool canJump(int* nums, int numsSize) {
    return canJumpFromPosition(0, nums, numsSize);
}

File: Editorial/0000-0100/0055_Jump_Game/JavaScript/1_Backtracking.js
var canJumpFromPosition = function (position, nums) {
    if (position == nums.length - 1) {
        return true;
    }
    let furthestJump = Math.min(position + nums[position], nums.length - 1);
    for (
        let nextPosition = position + 1;
        nextPosition <= furthestJump;
        nextPosition++
    ) {
        if (canJumpFromPosition(nextPosition, nums)) {
            return true;
        }
    }
    return false;
};
var canJump = function (nums) {
    return canJumpFromPosition(0, nums);
};
